### PR TITLE
Hacktoberfest Task 5

### PR DESCRIPTION
**Pull request by xMaxximum**
# Description

- Change the HasXy methods to flags (Enums)
- Create a method for the enum named HasFeature(GuildFeatureFlag)
- Create a method for the guild features to dynamically generate a string list of features

## Type of change

- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [x] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

@Aiko-IT-Systems/discatsharp

### CODE REVIEW

**Review by TheXorog — APPROVED** [review]

File: DisCatSharp/Entities/Guild/DiscordGuild.Features.cs [hunk 1]
```diff
@@ -21,298 +21,316 @@
...
 using System;
+using System.Collections.Generic;
 using System.Linq;
+using System.Text;
 
 using DisCatSharp.Enums;
 
 namespace DisCatSharp.Entities;
 
+public class GuildFeatures
```
Comment (Lulalaby) [inline DisCatSharp/Entities/Guild/DiscordGuild.Features.cs:32]: Please add an docstring header

File: DisCatSharp/Entities/Guild/DiscordGuild.Features.cs [hunk 2]
```diff
@@ -21,298 +21,316 @@
...
+{
+	public List<GuildFeaturesEnum> Features { get; }
```
Comment (Lulalaby) [inline DisCatSharp/Entities/Guild/DiscordGuild.Features.cs:34]: Add docs

File: DisCatSharp/Entities/Guild/DiscordGuild.Features.cs [hunk 3]
```diff
@@ -21,298 +21,316 @@
...
+
+		else
+		{
+			var humanReadableFeatures = this.Features.Select(x => AddSpacesToWord(x.ToString()));
+
+			return string.Join(separator, humanReadableFeatures);
+		}
+	}
+
+	private static string AddSpacesToWord(string text)
```
Comment (Lulalaby) [inline DisCatSharp/Entities/Guild/DiscordGuild.Features.cs:94]: Add docs

File: DisCatSharp/Entities/Guild/DiscordGuild.Features.cs [hunk 4]
```diff
@@ -21,298 +21,316 @@
...
+		if (guild.RawFeatures.Contains("LINKED_TO_HUB")) this.Features.Add(GuildFeaturesEnum.IsLinkedToHub);
+		if (guild.RawFeatures.Contains("MEMBER_PROFILES")) this.Features.Add(GuildFeaturesEnum.HasMemberProfiles);
+		if (guild.RawFeatures.Contains("INTERNAL_EMPLOYEE_ONLY")) this.Features.Add(GuildFeaturesEnum.IsStaffOnly);
+		if (guild.RawFeatures.Contains("ROLE_SUBSCRIPTIONS_AVAILABLE_FOR_PURCHASE")) this.Features.Add(GuildFeaturesEnum.RoleSubscriptionsIsAvailableForPurchase);
+		if (guild.RawFeatures.Contains("AUTO_MODERATION")) this.Features.Add(GuildFeaturesEnum.CanSetupAutoModeration);
+		if (guild.RawFeatures.Contains("GUILD_HOME_TEST")) this.Features.Add(GuildFeaturesEnum.GuildHomeTest);
+		if (guild.RawFeatures.Contains("INVITES_DISABLED")) this.Features.Add(GuildFeaturesEnum.InvitesDisabled);
+	}
+
+	public bool HasFeature(GuildFeaturesEnum flag) => this.Features.Contains(flag);
```
Comment (Lulalaby) [inline DisCatSharp/Entities/Guild/DiscordGuild.Features.cs:80]: Add docs

**Review by Lulalaby — CHANGES_REQUESTED** [review]
This is an really good outstanding pr in my opinion.
Congrats and thank you.

File: DisCatSharp/Entities/Guild/DiscordGuild.Features.cs [hunk 5]
```diff
@@ -31,288 +33,325 @@ namespace DisCatSharp.Entities;
...
+		else
+		{
+			var humanReadableFeatures = this.Features.Select(x => AddSpacesToWord(x.ToString()));
+
+			return string.Join(separator, humanReadableFeatures);
+		}
+	}
+
+	/// <summary>
+	/// Converts a string of characters (here: enum) into a string of characters seperated by spaces after a capital letter.
```
Comment (Lulalaby) [inline DisCatSharp/Entities/Guild/DiscordGuild.Features.cs:111]:
```suggestion
	/// Converts a string of characters (here: enum) into a string of characters separated by spaces after a capital letter.
```

File: DisCatSharp/Entities/Guild/DiscordGuild.Features.cs [hunk 6]
```diff
@@ -31,288 +33,325 @@ namespace DisCatSharp.Entities;
...
+	/// </summary>
+	/// <param name="text">String of text to convert</param>
+	/// <returns>String seperated by a space after every capital letter.</returns>
```
Comment (Lulalaby) [inline DisCatSharp/Entities/Guild/DiscordGuild.Features.cs:114]:
```suggestion
	/// <returns>String separated by a space after every capital letter.</returns>
```

**Review by Lulalaby — APPROVED** [review]
LGTM

**Pull request merged**